src/services/WorkspaceArtifactsApi.ts

```diff
@@ -0,0 +1,37 @@
+import { Artifact, ArtifactResponse } from '@/models'
+import { ArtifactFilter } from '@/models/Filters'
+import { mapper } from '@/services/Mapper'
+import { WorkspaceApi } from '@/services/WorkspaceApi'
+
+export interface IWorkspaceArtifactsApi {
+  getArtifact: (id: string) => Promise<Artifact>,
+  getArtifacts: (filter: ArtifactFilter) => Promise<Artifact[]>,
+  getArtifactsCount: (filter: ArtifactFilter) => Promise<number>,
+  deleteArtifact: (id: string) => Promise<void>,
+}
+
+export class WorkspaceArtifactsApi extends WorkspaceApi implements IWorkspaceArtifactsApi {
+
+  protected override routePrefix = '/artifacts'
+
+  public async getArtifact(id: string): Promise<Artifact> {
+    const { data } = await this.get<ArtifactResponse>(`/${id}`)
+    return mapper.map('ArtifactResponse', data, 'Artifact')
```

Comment:
Should we use the batch processor here?

Reply:
I don't exactly have a use for it right now but... why not! Will test it a bunch while working on this

Reply:
Good enough reason for me 👍

Reply:
updated - nice suggestion